// hyper-darkmatter/main.js
"use strict";
var assign, backgroundColor, blue, borderColor, cursorColor, cyan, decorateConfig, foregroundColor, green, lightBlack, lightBlue, lightCyan, lightGreen, lightMagenta, magenta, red, yellow;

assign = Object.assign;

backgroundColor = "#1c1f26";

foregroundColor = "#c0c5ce";

borderColor = backgroundColor;

cursorColor = foregroundColor;

red = "#d5713f";

green = "#75bb36";

yellow = "#e7e781";

blue = "#267faf";

magenta = "#a76ebf";

cyan = "#22bae4";

lightCyan = "#58dbff";

lightGreen = "#90cc5b";

lightBlack = "#65737e";

lightBlue = blue;

lightMagenta = "#d1a9ce";

decorateConfig = function(config) {
  return assign({}, config, {
    backgroundColor: backgroundColor,
    foregroundColor: foregroundColor,
    cursorColor: cursorColor,
    borderColor: borderColor,
    colors: {
      black: backgroundColor,
      white: foregroundColor,
      red: red,
      green: green,
      blue: blue,
      cyan: cyan,
      magenta: magenta,
      yellow: yellow,
      lightRed: red,
      lightYellow: yellow,
      lightMagenta: lightMagenta,
      lightBlack: lightBlack,
      lightWhite: foregroundColor,
      lightCyan: lightCyan,
      lightBlue: lightBlue,
      lightGreen: lightGreen
    },
    css: config.css + "\n.tab_active {\n  background: rgba(101, 115, 126, .10)\n}"
  });
};

module.exports.decorateConfig = decorateConfig;
